libs/engine_threading.py

```diff
@@ -44,15 +44,20 @@ def __init__(self, config, source, live_feed_enabled=True):
 
     def run(self):
         self.engine = CvEngine(self.config, self.source['section'], self.live_feed_enabled)
+        restarts = 0
+        max_restarts = int(self.config.get_section_dict("App")["MaxThreadRestarts"])
         while True:
             try:
                 self.engine.process_video(self.source['url'])
             except Exception as e:
                 logging.error(e, exc_info=True)
+                logging.info(f"Exception processing video for source {self.source['name']}")
+                if restarts == max_restarts:
```

Comment:
`restarts` never gets reset so this is counting restarts no matter how much time passed between one and the next (could be a week for example). Not sure we need to go so deep and improve that but we could (save time of last restart and if it is greater than 1/2/3 minutes then reset count to 0)

Reply:
Yes, I wasn't sure how much complexity to include in the logic. But make sense to take into account the time variable. I will include it.

Reply:
Done